### DISCUSSION TITLE
Can't use ItemStack Extraction Upgrade with Extractor Module

### DISCUSSION THREAD

**Discussion by Mandaroni** [RS485/LogisticsPipes#1710] [Q&A]
Can't use "ItemStack Extraction Upgrade" with Extractor Module although Tooltip says its compatible. It doesn't let me put the upgrades in the pipe. Item Speed Upgrade works.

**Comment by bziemons** [comment]
You need the module upgrade in the chassis pipe to be able to add upgrades to modules. The upgrade slots show up in the regular chassis UI.
